### PR TITLE
SA-1587: Delete a saved report

### PR DESCRIPTION
**Pull request by kchu1337**
### What Changed
 - Added delete action/reducer
 - Added delete modal
 - Added tests
 - Fixed some older bugs
### How To Test
- Go to report builder in Hibana. Enable saved reports with UI user option `allow_saved_reports`.
- Make a new saved report through the save new report button.
- Once that succeeds, click the view all reports button.
- Find your report in the table. Click on the actionlist (...) and then click delete.
- Confirm the deletion by clicking Delete in the new modal.

### CODE REVIEW

File: src/pages/reportBuilder/components/SavedReportsSection/ReportsListModal.js [hunk 1]
```diff
@@ -42,14 +51,47 @@ export function ReportsListModal(props) {
...
-        <Button.Icon as={MoreHoriz} />
-      </Button>,
+      <Popover
+        left
+        id={`popover-myReports-${id}`}
+        trigger={
+          <Button
+            variant="minimal"
+            aria-controls={`popover-myReports-${id}`}
+            data-id={`myReports-${id}`}
```
Comment (kchu1337) [inline src/pages/reportBuilder/components/SavedReportsSection/ReportsListModal.js:61]: IDK if it needs all these props but I was just making changes until my Cypress tests would successfully open the delete modal.

File: src/pages/reportBuilder/components/SaveNewReportModal.js [hunk 2]
```diff
@@ -25,13 +25,15 @@ export function SaveNewReportModal(props) {
     return createReport({ ...data, query_string }).then(() => {
       showAlert({ type: 'success', message: `You have successfully saved ${data.name}` });
       onCancel();
+      getReports();
```
Comment (kchu1337) [inline src/pages/reportBuilder/components/SaveNewReportModal.js:28]: Added this so the list of reports reloads with the newly added report.

File: src/pages/reportBuilder/components/SavedReportsSection/ReportsListModal.js [hunk 3]
```diff
@@ -46,14 +55,47 @@ export function ReportsListModal(props) {
...
+        <ActionList actions={[{ content: `Delete`, onClick: () => handleDelete({ id, name }) }]} />
+      </Popover>,
     ];
   };
 
   const allReportsRows = report => {
-    const { name, modified, creator, subaccount_id } = report;
+    const { id, name, modified, creator, subaccount_id, current_user_can_edit } = report;
+    //conditionally render the actionlist
+    const action = current_user_can_edit
```
Comment (PatrickSison) [inline src/pages/reportBuilder/components/SavedReportsSection/ReportsListModal.js:80]: Small question, but does this need to be an array? Not an issue, just curious

File: src/pages/reportBuilder/components/SavedReportsSection/ReportsListModal.js [hunk 4]
```diff
@@ -46,14 +55,47 @@ export function ReportsListModal(props) {
...
+            variant="minimal"
+            aria-controls={`popover-myReports-${id}`}
+            data-id={`myReports-${id}`}
+          >
+            <Button.Icon as={MoreHoriz} />
+            <ScreenReaderOnly>Open Menu</ScreenReaderOnly>
+          </Button>
+        }
+      >
+        <ActionList actions={[{ content: `Delete`, onClick: () => handleDelete({ id, name }) }]} />
```
Comment (PatrickSison) [inline src/pages/reportBuilder/components/SavedReportsSection/ReportsListModal.js:72]: This is also a deprecated way of adding actions to ActionList now. I think new way is through ActionList.Action

File: src/pages/reportBuilder/components/SavedReportsSection/ReportsListModal.js (hunk 3 above)
Comment (kchu1337) [reply src/pages/reportBuilder/components/SavedReportsSection/ReportsListModal.js:80]: Just so i can use the spread notation, but I don't think it has to be an array.

File: src/pages/reportBuilder/components/SavedReportsSection/SavedReportsSection.js [hunk 5]
```diff
@@ -79,13 +79,20 @@ const SavedReportsSection = props => {
...
-        content={`Are you sure you want to delete ${toDelete.name}`}
+        content={
+          <p>
+            Are you sure you want to delete <Bold>"{toDelete.name}"</Bold>
+          </p>
+        }
         open={modalStatus === 'delete'}
-        isPending={props.isDeletePending}
-        onCancel={() => setModalStatus('')}
+        isPending={props.isDeletePendiKng}
```
Comment (PatrickSison) [inline src/pages/reportBuilder/components/SavedReportsSection/SavedReportsSection.js:92]: Typo, should be `isDeletePending`

**Review by PatrickSison — APPROVED** [review]
Just small fix, looks good otherwise

**Pull request merged**